vmdb/app/controllers/application_controller/ci_processing.rb

```diff
@@ -1897,14 +1897,14 @@ def owner_changed?(owner)
     @edit[:new][owner] != @edit[:current][owner]
   end
 
-  def show_association(action, display_name, listicon, method, klass, association = nil, conditions =nil)
+  def show_association(action, display_name, listicon, method, klass, association = nil, conditions = nil, record = nil)
     # Ajax request means in explorer, or if current explorer is one of the explorer controllers
     @explorer = true if request.xml_http_request? && explorer_controller?
     if @explorer  # Save vars for tree history array
       @x_show = params[:x_show]
       @sb[:action] = @lastaction = action
     end
-    @record = identify_record(params[:id])
+    @record = record.present? ? record : identify_record(params[:id])
```

Comment:
@himdel i am confused my understanding is that params[:id] is present when show_association is called, shouldn't that be enough why do we have to pass in record to this method.